simSPI/fov.py

```diff
@@ -0,0 +1,249 @@
+
+    x_range = np.arange(x_origin, x_frontier, boxsize)
+    y_range = np.arange(y_origin, y_frontier, boxsize)
+    n_particles = np.int(fov_nx * fov_ny)
+    return x_range, y_range, n_particles
+
+
+def get_fov(optics_params, detector_params, pdb_file=None, dmax=100, pad=1.0):
+    """Define field of view dimensions for displaying particle.
+
+    Parameters
+    ----------
+    optics_params : list
+        List of sim parameters pertaining to microscope settings.
+    detector_params : list
+        List of sim parameters pertaining to detector settings.
+    pdb_file : str
+        Relative path to .pdb file output.
+    dmax : int
+        Maximum dimension of molecule.
+    pad : int
+        Amount of padding.
+
+    Returns
+    -------
+    fov_Lx : float
+        Length of fov in x-dimension.
+    fov_Ly : float
+        Length of fov in y-dimension.
+    boxsize : float
+        Boxsize of particle, equal to max particle dimension with pad.
+    """
+    detector_nx_pix = detector_params[0]
+    detector_ny_pix = detector_params[1]
+    detector_pix_size = detector_params[2] * 1e3
+    magnification = optics_params[0]
+
+    detector_lx = detector_nx_pix * detector_pix_size
+    detector_ly = detector_ny_pix * detector_pix_size
+    fov_lx = detector_lx / magnification
+    fov_ly = detector_ly / magnification
+
+    if pdb_file is not None:
+        dmax = get_dmax(pdb_file)
+
+    boxsize = dmax + 2 * pad
+    return fov_lx, fov_ly, boxsize
+
+
+def get_dmax(filename):
+    """Get maximmum dimension of particle.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+
+    Returns
+    -------
+    np.amax(distance) : float
+        Max dimension of given particle in .pdb source file.
+    """
+    xyz = get_xyz_from_pdb(filename)
+    distance = pdist(xyz[0, ...])
+    dmax = np.amax(distance)
+
+    return dmax
+
+
+def get_xyz_from_pdb(filename=None, atom_selection="name CA or name P"):
+    """Get particle coordinates from .pdb file.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+    atom_selection: str
+        Atoms to be selected by MDTraj for returning their cartesian coordinates
+
+    Returns
+    -------
+    traj_small.xyz : ndarray
+        Particle coordinates from .pdb file.
+    """
+    traj = md.load(filename)
+    atom_indices = traj.topology.select(atom_selection)
+    traj_small = traj.atom_slice(atom_indices)
+    coordinates = traj_small.xyz
+
+    return coordinates
+
+
+def micrograph2particles(
+    micrograph, optics_params, detector_params, pdb_file=None, dmax=30, pad=5.0
+):
+    """Extract particles from given micrograph.
+
+    Parameters
+    ----------
+    micrograph : ndarray
+        Particle micrograph to extract from.
+    optics_params : list
+        List of sim parameters pertaining to microscope settings.
```

Comment:
Replace by `Simulations parameters related to microscope settings`
(avoid using the abbreviation sim, and we don't need to specify that it is a list, it is said above)

To be changed here and elsewhere.